rclcpp/src/rclcpp/node_interfaces/node_base.cpp

```diff
@@ -115,13 +112,30 @@ NodeBase::NodeBase(
     throw_from_rcl_error(ret, "failed to initialize rcl node");
   }
 
+  // The initialization for the rosout publisher
+  if (rcl_logging_rosout_enabled() && rcl_node_options.enable_rosout) {
+    std::lock_guard<std::recursive_mutex> guard(*logging_mutex);
+    ret = rcl_logging_rosout_init_publisher_for_node(rcl_node.get());
+    if (ret != RCL_RET_OK) {
+      throw_from_rcl_error(ret, "failed to initialize rosout publisher");
+    }
+  }
+
   node_handle_.reset(
     rcl_node.release(),
     [logging_mutex](rcl_node_t * node) -> void {
-      std::lock_guard<std::recursive_mutex> guard(*logging_mutex);
-      // TODO(ivanpauno): Instead of mutually excluding rcl_node_fini with the global logger mutex,
-      // rcl_logging_rosout_fini_publisher_for_node could be decoupled from there and be called
-      // here directly.
+      const rcl_node_options_t * node_ops = rcl_node_get_options(node);
```

Comment:
`node_ops` will not be NULL here.

we can do something else like, declare bool flag to set to true in default. and if `node_ops` is not null, store that value to it then call `rcl_logging_rosout_fini_publisher_for_node`. but that seems to be overkill since `node_ops` will not be NULL here, what do you think?

Reply:
@iuhilnehc-ynos any thoughts?